src/modules/local_position_estimator/BlockLocalPositionEstimator.cpp

```diff
@@ -1143,15 +1344,37 @@ void BlockLocalPositionEstimator::correctLidar()
 	// kalman filter correction if no fault
 	if (_lidarFault == FAULT_NONE) {
 		Matrix<float, n_x, n_y_lidar> K = _P * C.transpose() * S_I;
-		_x += K * r;
+		Vector<float, n_x> dx = K * r;
```

Comment:
The small amount of drift in xy was a result of the correction for sonar/ baro/ lidar. This ensures that if xy can't be measured that the altitude measurements don't inject small amounts of error in xy that eventually add up to  a large value.